### PR TITLE
Fix/not changing the switched chain id in in app browser

### PR DESCRIPTION
**Pull request by q20274982**
## Summary

by adding eventListeners to the `bloctoSDK` and also adding it in the in-app browser's existedSDK,
it allows invoking event functions in the in-app-browser as well.

## Related Links

**Asana**: https://app.asana.com/0/1201812548509877/1205706441564006/f

## Checklist

- [x] Pasted Asana link.
- [x] Tagged labels.

### CODE REVIEW

File: packages/blocto-sdk/src/providers/ethereum.ts [hunk 1]
```diff
@@ -907,4 +911,20 @@ export default class EthereumProvider
       throw ethErrors.rpc.invalidParams('Empty networkList');
     }
   }
+
+  override on(event: string, listener: (arg: any) => void): void {
+    if (this.existedSDK?.isBlocto)
```
Comment (q20274982) [inline packages/blocto-sdk/src/providers/ethereum.ts:916]: I think the condition `if (this.existedSDK?.isBlocto)` is more concise and achieves the same effect as `const existedSDK = (window as any).ethereum; if (existedSDK && existedSDK.isBlocto)`. If there are no issues, I'll replace `const existedSDK = (window as any).ethereum; if (existedSDK && existedSDK.isBlocto)` with it.

**Review by sanyu1225 — APPROVED** [review]

File: packages/blocto-sdk/src/providers/ethereum.ts (hunk 1 above)
Comment (q20274982) [reply packages/blocto-sdk/src/providers/ethereum.ts:914]: I replaced all the `const existedSDK = (window as any).ethereum; if (existedSDK && existedSDK.isBlocto)` with `if (this.existedSDK?.isBlocto)` in this commit [refactor: enhance existedSDK check condition](https://github.com/blocto/blocto-sdk/pull/288/commits/44c70c71a73833ca36881ec84981b38deca663a8)

**Review by sanyu1225 — APPROVED** [review]

**Review by mordochi — COMMENTED** [review]
這個 proxy event 的操作可以也幫我加到 aptos provider 嗎？

**Comment by q20274982** [comment]
> 這個 proxy event 的操作可以也幫我加到 aptos provider 嗎？

solved here [feat: enable aptos's existedSDK event](https://github.com/blocto/blocto-sdk/pull/288/commits/51a3a80dc893c9a6932e6c41dc0f436051c73c29)

**Review by mordochi — APPROVED** [review]
如果 local 測完都沒問題（adapters 也試試看） 就可以幫我 merge 然後發一版 beta 版給 @scottphc 用用看ㄛ

**Pull request merged**